source/utils/Random.cpp

```diff
@@ -36,58 +30,35 @@
 unsigned long myRandomSeed;
 const unsigned long MAX = 32768;
 
-unsigned long randgen()
+static unsigned long randgen()
 {
     myRandomSeed = myRandomSeed * 1103515245 + 12345;
-    unsigned long returnVal = (unsigned int) (myRandomSeed / 65536) % MAX;
+    //TODO: What is the purpose of the cast?
```

Comment:
I can see 2 possible reasons:
- To strip the high bits. Maybe is it better for the algorithm to have only 32 bits (on most architectures)
- returnVal used to be an unsigned int and there was a cast to avoid a warning. Then, when returnVal  became unsigned long, because no warning was triggered from smaller to bigger type, it stayed unseen.